### ISSUE TITLE
[firebase_core] FirebaseInitProvider ClassNotFoundException on old Android versions

### ISSUE THREAD

**Issue by viniciuszani** [FirebaseExtended/flutterfire#1340]
**Bug description**
Firebase causes old devices to crash the application during its initialization process.
Device which leads to the error:
- Model Name: Galaxy S3
- RAM: 1024 MB
- Manufacturer: Samsung
- OpenGL ES Version: 2.0
- Android Version: Android 4.3
- Native Platform: armeabi-v7a
- Locale: en_US

**To Reproduce**
1. Create a simple flutter application with `firebase_core: ^0.4.0+9` (and all the steps required to integrate the library into the project)
2. Build the release version of the application
3. Run it on a device with similar specs as the ones listed above (armeabi-v7a / Android 4.x)

**Expected behavior**
The application should crash when opened, and the stacktrace should be the following:

```
FATAL EXCEPTION: main
java.lang.RuntimeException: Unable to get provider com.google.firebase.provider.FirebaseInitProvider: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.provider.FirebaseInitProvider" on path: /data/app/com.redacted-1.apk
	at android.app.ActivityThread.installProvider(ActivityThread.java:5122)
	at android.app.ActivityThread.installContentProviders(ActivityThread.java:4725)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4665)
	at android.app.ActivityThread.access$1400(ActivityThread.java:159)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1376)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:176)
	at android.app.ActivityThread.main(ActivityThread.java:5419)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:525)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1046)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:862)
	at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.provider.FirebaseInitProvider" on path: /data/app/com.redacted-1.apk
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:64)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:501)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:461)
	at android.app.ActivityThread.installProvider(ActivityThread.java:5107)
	... 12 more
```

**Comment by Qamhan** [comment]
same problem i fixed it by doing this change [#21009](https://github.com/flutter/flutter/issues/21009) in my project.
do this only if enabled multiDex in your project.

**Comment by vinicius-animo** [comment]
Hi @Qamhan, I have applied the solution and it looks like it did work. Thanks!
